### PR TITLE
feat(10): Add working FB auth and login

### PR DESCRIPTION
**Pull request by hromogen**
- Add working backend implementation for passport facebook login (via passport-facebook-token)
- Add and style the button for initial facebook auth from front
- Add handle facebook response at front (bind front with back end login)
- Update/refactor unit tests at front
- Add/refactor unit tests at back

### CODE REVIEW

File: packages/web/src/components/Container/Container.tsx [hunk 1]
```diff
@@ -1,11 +1,11 @@
 import React from 'react';
+import classnames from 'classnames';
```
Comment (dubbha) [inline packages/web/src/components/Container/Container.tsx]: camelCase is commonly used and listed in docs examples of the lib itself:
```
import classNames from  'classnames';
...
<div className={classNames(['container', className])}>{children}</div>
```

File: packages/web/src/pages/SignIn/SignIn.tsx [hunk 2]
```diff
@@ -19,10 +20,18 @@ export const SignIn = () => {
   return (
     <Container>
       <Header />
-      <SignInForm isLoading={isLoading} error={error} onSubmit={handleSubmit} />
+      <Container className="signInConrollersContainer">
```
Comment (dubbha) [inline packages/web/src/pages/SignIn/SignIn.tsx]: Container component was supposed to be used on the page top level only. Just group components into div or extract them, no need to re-use the Component with modified styling.

File: packages/web/src/components/SignInForm/styles.scss [hunk 3]
```diff
@@ -4,9 +4,12 @@
...
   width: 70%;
-  margin-bottom: 76px;
 
   .register {
     margin-top: 16px;
   }
 }
+
+.signinForm {
+  margin-bottom: 20px !important;
```
Comment (karolinayanko) [inline packages/web/src/components/SignInForm/styles.scss]: can we avoid !important and use more complex selector instead?

File: packages/web/src/pages/SignIn/SignIn.tsx (hunk 2 above)
Comment (karolinayanko) [reply packages/web/src/pages/SignIn/SignIn.tsx]: Agree, lets keep Container as page wrapper only

File: packages/web/src/pages/Profile/__snapshots__/Profile.test.tsx.snap [hunk 4]
```diff
@@ -8,7 +8,7 @@ exports[`pages/Profile should render successfully if user is logged in 1`] = `
     Header
   </div>
   <div>
-    NAME [EMAIL]
+     [EMAIL]
```
Comment (oleksiydubovykTR) [inline packages/web/src/pages/Profile/__snapshots__/Profile.test.tsx.snap]: Where did the name disappear?

File: packages/web/src/components/SignInForm/styles.scss [hunk 5]
```diff
@@ -4,9 +4,12 @@
...
-  margin-bottom: 76px;
 
   .register {
     margin-top: 16px;
   }
 }
+
+form.signinForm {
+  margin-bottom: 20px;
+}
```
Comment (oleksiydubovykTR) [inline packages/web/src/components/SignInForm/styles.scss:15]: Why did you remove my 76px margin? And why is it now 20px?
The idea was that it compensated for a notification placeholder above, so it doesn't jump and  it vertically centered.
Is it for the additional facebook button?

File: packages/web/src/pages/Profile/__snapshots__/Profile.test.tsx.snap (hunk 4 above)
Comment (oleksiydubovykTR) [reply packages/web/src/pages/Profile/__snapshots__/Profile.test.tsx.snap]: Profile.test should probably have been updated instead of the snaphost

File: packages/web/src/components/SignInForm/styles.scss (hunk 5 above)
Comment (hromogen) [reply packages/web/src/components/SignInForm/styles.scss:15]: Yep. I have tried wrong email for this page and after notification have not noticed any "jumps". Mb new container (i.e. `div className="container ..`) solves the problem

**Review by karolinayanko — APPROVED** [review]

File: packages/web/src/pages/SignIn/SignIn.tsx [hunk 6]
```diff
@@ -1,28 +1,39 @@
...
-      payload: { email, password },
+      payload: { email, password }
     });
   };
 
   return (
     <Container>
       <Header />
-      <SignInForm isLoading={isLoading} error={error} onSubmit={handleSubmit} />
+      <div className="container signInConrollersContainer">
```
Comment (dubbha) [inline packages/web/src/pages/SignIn/SignIn.tsx]: You're still using a `container` class which comes from Bootstrap, and has hidden behaviours, e.g. mediaQueries. Now we need to remember about this, in case it start to behave weirdly.

**Review by dubbha — APPROVED** [review]

File: packages/server/src/utils/create-tokenized-user.ts [hunk 7]
```diff
@@ -0,0 +1,18 @@
...
+
+export function createTokenizedUser(user: UserEntity): UserWithToken {
+  const tokenData = createToken(user);
+  const { id, firstName, lastName, email, status, facebookId } = user;
+  return {
+    id,
+    firstName,
+    lastName,
+    email,
+    status,
```
Comment (dartwlad) [inline packages/server/src/utils/create-tokenized-user.ts]: Do we really need to send user status to client?

File: packages/web/src/pages/SignIn/SignIn.tsx (hunk 6 above)
Comment (dartwlad) [reply packages/web/src/pages/SignIn/SignIn.tsx]: It looks like you should use row instead.

File: packages/web/src/pages/SignIn/__snapshots__/SignIn.test.tsx.snap [hunk 8]
```diff
@@ -89,67 +89,72 @@ exports[`pages/SignIn should render successfully 1`] = `
       </div>
     </nav>
   </header>
-  <form
-    class=""
+  <div
+    class="container signInConrollersContainer"
```
Comment (dartwlad) [inline packages/web/src/pages/SignIn/__snapshots__/SignIn.test.tsx.snap]: Same here. 
And just a question, do we use camelCase naming for CSS classes?

Comment (hromogen) [reply packages/web/src/pages/SignIn/__snapshots__/SignIn.test.tsx.snap]: I don't see what is wrong with camelCase in CSS

**Review by dartwlad — APPROVED** [review]
